03_014/frontend/src/core/Component.ts

```diff
@@ -0,0 +1,127 @@
+import { ComponentOptions, Props, State, ComponentStoreOptions, SetStateOptions } from "@src/utils/types";
+import { Publisher, Subscriber } from "@src/core/Store";
+
+class Component {
+  $target: Element;
+  state: State = {};
+  props: Props = {};
```

Comment:
State, Props 타입을 고정적으로 관리하기보단, Generic type을 이용해보면 어떨까요?

```suggestion
class Component<State = {}, Props = {}> {
  $target: Element;
  state: State = {};
  props: Props = {};
```
이렇게요!